utils/large_bitset.cc

```diff
@@ -17,15 +17,13 @@ using namespace seastar;
 large_bitset::large_bitset(size_t nr_bits) : _nr_bits(nr_bits) {
     assert(thread::running_in_thread());
 
-    const size_t orig_nr_ints = align_up(nr_bits, bits_per_int()) / bits_per_int();
-    auto nr_ints = orig_nr_ints;
-    while (nr_ints) {
-        nr_ints = _storage.reserve_partial(nr_ints);
+    size_t nr_ints = align_up(nr_bits, bits_per_int()) / bits_per_int();
+    while (_storage.capacity() != nr_ints) {
+        _storage.reserve_partial(nr_ints);
         if (need_preempt()) {
             thread::yield();
         }
     }
```

Comment:
Suggest to add (in a separate patch) reserve_gently(gently_reserveable auto& container, size_t).  See utils/stall_free.hh.

Reply:
Sure. I'll add as a follow up to this PR.

Reply:
reserve_gently PR at https://github.com/scylladb/scylladb/pull/19361